src/ensembl/src/content/app/browser/hooks/useBrowserRouting.ts

```diff
@@ -115,25 +115,26 @@ const useBrowserRouting = () => {
        because it will also try to bookmark the Ensembl object that is stored in redux state
        before it gets changed by setDataFromUrlAndSave
       */
-      dispatch(changeFocusObject(newFocusId as string));
+      changeFocusObject(newFocusId as string);
     } else if (chrLocation) {
       const isSameLocationAsInRedux =
         activeGenomeId && isEqual(chrLocation, allChrLocations[activeGenomeId]);
       const isFirstRender = firstRenderRef.current;
 
-      if (!isSameLocationAsInRedux || isFirstRender) {
-        dispatch(
-          changeBrowserLocation({
-            genomeId,
-            ensObjectId: newFocusId,
-            chrLocation
-          })
-        );
+      if ((!isSameLocationAsInRedux || isFirstRender) && genomeBrowser) {
+        changeFocusObject(newFocusId as string);
+
+        changeBrowserLocation({
+          genomeId,
+          ensObjectId: newFocusId,
+          chrLocation
+        });
```

Comment:
I'm lost. You are both passing `ensObjectId` parameter to `changeBrowserLocation`, and also calling `changeFocusObject` Seems excessive.